src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -672,23 +707,39 @@ public static void createAndAddNewObjectMap(@NonNull Object id, List<Map<String,
         objectList.add(newObjectMap);
     }
 
+    public static Map<String, Object> getObjectsMapHavingClient(RestHighLevelClient client, String index, String id, String object) throws Exception {
+        Map<String, Object> objectsMap = null;
+        if (client != null) {
+            objectsMap = getObjectsMap(client, index, id, object);
+            return objectsMap;
+        }
+        else {
+            logger.error("RestHighLevelClient was null");
+        }
+        return objectsMap;
+    }
+
+    public static Map<String, Object> getObjectsMap(RestHighLevelClient client, String index, String id, String object) throws Exception {
+        String[] includes = new String[] {object};
+        String[] excludes = Strings.EMPTY_ARRAY;
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, includes, excludes);
+        GetRequest getRequest = new GetRequest()
+                .index(index)
+                .type("_doc")
+                .id(id)
+                .fetchSourceContext(fetchSourceContext);
+
+        GetResponse getResponse = null;
+        if (client.exists(getRequest, RequestOptions.DEFAULT)) {
+            getResponse = client.get(getRequest, RequestOptions.DEFAULT);
+        }
+        return getResponse != null ? getResponse.getSourceAsMap() : null;
```

Comment:
Might avoid client NPEs by returning `Collections.emptyMap()` instead of `null`.